CMakeLists.txt

```diff
@@ -66,6 +66,8 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}
   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     add_compile_options(-Wno-multichar -Wno-sign-compare)
   endif ()
+elseif ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "Clang")
+  add_compile_options(-Wall -Werror -Wmissing-prototypes)
```

Comment:
I think this block will never be used. We already have an almost identical check at line 64. You wanna put `-Wmissing-prototype` at line 65.

Reply:
The difference is whether we are invoking a C compiler or a C++ compiler. There are some libraries in this project which invoke the C compiler and CMake does not append C++ flags to invocations of the C compiler.
We cannot add -Wmissing-prototype to C++ CFLAGS because there are no "prototypes" in C++ and g++ will warn us which in our case with -Werror means those are errors.

Reply:
While it does need to distinguish whether this is a C compiler or a C++ compiler, I see that  add_compile_options() may not help with this discernment, I'm reviewing CMake documentation, I'll check to see if this is the correct approach.

Reply:
This is resolved with [this commit](https://github.com/facebook/xcbuild/pull/312/commits/32e37e9047c51deb25572ffd53331fab9bc83085).